[MDS-5431] create new or historical record from one window - in New Flow

isuru-aot opened the pull request:
## Objective 

[MDS-5431](https://bcmines.atlassian.net/browse/MDS-5431)

_Why are you making this change? Provide a short explanation and/or screenshots_

This is part of the [PR-2708](https://github.com/bcgov/mds/pull/2708). With this change the same flow is enabled in the new flow (when the `esup_permit_amendment` feature is enabled).

isuru-aot committed:
basic new modal added

isuru-aot committed:
Merge branch 'develop' of https://github.com/bcgov/mds into mds-5431-create-new-or-historical-record-from-one-window

isuru-aot committed:
completed flow with html tags

isuru-aot committed:
adding typography

isuru-aot committed:
formtting

isuru-aot committed:
formtting

isuru-aot committed:
formtting

isuru-aot committed:
fixing tests and styling

isuru-aot committed:
pr comments addressed + feature flag

isuru-aot committed:
fixing failing tests

isuru-aot committed:
typo correction

isuru-aot committed:
fixing tests

isuru-aot committed:
resolving merge conflicts

isuru-aot committed:
adding 5431 feature to new flow added in ExplosivePermitFormNew.tsx

taraepp commented on services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitForm.tsx:
```diff
@@ -117,13 +119,119 @@ export const ExplosivesPermitForm: FC<ExplosivesPermitFormProps &
+      <Form layout="vertical">
+        <Typography.Title level={3}>Add Permit</Typography.Title>
+        <div>
+          <Typography.Paragraph>Let's get your permit started, in CORE you can...</Typography.Paragraph>
```
I just ran into this with my PR so I gotta check- was this pushed as --no-verify? If so, can it be linted? Generally it doesn't like unescaped ' and " so that was the tip-off.

isuru-aot replied:
Updated. Please verify.

taraepp commented on services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitFormNew.tsx:
```diff
@@ -119,11 +121,117 @@ export const ExplosivesPermitFormNew: FC<ExplosivesPermitFormNewProps> = (props)
+  const handleRadioChange = (e) => {
+    setRadioSelection(e.target.value);
+    setIsHistoric(e.target.value == 1);
+    setIsAmend(e.target.value==3);
```
yeeeeah please lint these files. For the PRs of the future.

isuru-aot replied:
Updated. Please verify.

isuru-aot committed:
fixing formatting

matbusby-fw commented on services/core-web/src/components/Forms/ExplosivesPermit/ExplosivesPermitForm.tsx:
```diff
@@ -117,13 +119,119 @@ export const ExplosivesPermitForm: FC<ExplosivesPermitFormProps &
+  );
+
   return (
+    isFeatureEnabled(Feature.ONE_WINDOW_FOR_CREATING_NEW_OR_HISTORICAL_ESUP) && parentView ? (
```
While it doesn't hurt anything to have this included here as well as the `ExplosivesPermitFormNew`, Once it's live we'll eventually be deleting this file (and renaming the new one to this name), so it won't provide much benefit.

isuru-aot replied:
Agreed.  Only thing is, with this both changes (5353 & 5431) can be controlled using two different flags.
WDYT? Removing this and controlled both features through one flag is better?

matbusby-fw replied:
I lean towards a single flag since it's all part of the same feature that will be released as a whole.

isuru-aot replied:
Applied the changes needed for this. Please verify.

isuru-aot committed:
linting

isuru-aot committed:
resolving merge conflict

isuru-aot committed:
using ESUP_PERMIT_AMENDMENT feature flag

isuru-aot committed:
merging + feature flag changed

asinn134 committed:
[MDS-5456] submission documents not filtering out deleted_ind == true records (#2715)

* adding soft deletion of now_application_document_identity_xref

* adding soft delete in now_submissions document table

* adding document.deleted_ind check for false in now_app

* filtering out now_submission docs with deleted_ind equal true

* adding deleted_ind check to submission_documents secondary join

* removing space change

* removing space change

* removing space change

henryoforeh-dev committed:
[MDS-4259] ESUP permit Amendment without issuing new permit (#2716)

remove update on issue date.

matbusby-fw reviewed (dismissed):
🙌🏻

matbusby-fw committed:
[MDS-5353] -Refactor Explosives Permit Form for better readability and functionality (#2714)

* Refactor Explosives Permit Form for better readability and functionality

The code refactor has mainly focused on the `MagazineFormNew.tsx` and `ExplosivesPermitFormNew.tsx` components. Unnecessary `Form.Item` wrappers have been removed, resulting in cleaner code and more readable form fields. Ability to remember and set active panel for the Collapse element in the MagazineForm has been added which will enhance user experience as users will no longer lose their panel selections after a page reload. In the `ExplosivesPermitFormNew.tsx`, the form name has been made more descriptive and specific (EXPLOSIVES_PERMIT_NEW). Also, removed unused 'bindActionCreators and change' import to enhance performance. All these changes were made based on PR review comments.

* made delete button clickable

* fixed a couple of file errors

* remove broken validation

* restored Tara's work

simensma-fresh committed:
[MDS-5337] Cache fontawesome using Artifactory (#2712)

* switched compare function to moment from native Date so that 2023-02-15 does not become 2023-02-14 at 6pm local time

* [MDS-5337] Use artifactory to cache fontawesome pacakges

* Updated yarn.lock

* Fixed react type errors

---------

Co-authored-by: Tara Epp <102187683+taraepp@users.noreply.github.com>

simensma-fresh committed:
[MDS-5537] Fix fontawesome auth issue if you were previously authenticated (#2717)

* [MDS-5337] Use artifactory to cache fontawesome pacakges

* Updated yarn.lock

* Fixed react type errors

* MDS-5537 Unset old fontawesome token config

isuru-aot committed:
resolving merge conflict

isuru-aot committed:
Merge branch 'mds-5431-create-new-or-historical-record-from-one-window' of https://github.com/bcgov/mds into mds-5431-create-new-or-historical-record-from-one-window

isuru-aot commented:
Create a separate PR for this. https://github.com/bcgov/mds/pull/2720
